src/openrct2/world/Map.cpp

```diff
@@ -2918,6 +2979,10 @@ void tile_element_remove(rct_tile_element *tileElement)
         } while (!tile_element_is_last_for_tile(++tileElement));
     }
 
+    if (tileElement == nullptr)
```

Comment:
I don't see how this one would be hit. Shouldn't the check be at the start of the function?